cmd/orphan-finder/main.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/rpc"
+)
+
+type config struct {
+	AMQP   cmd.AMQPConfig
+	Statsd cmd.StatsdConfig
+	Syslog cmd.SyslogConfig
+}
+
+var (
+	b64derOrphan = regexp.MustCompile(`b64der=\[([a-zA-Z0-9+/]+)\]`)
+	regOrphan    = regexp.MustCompile(`regID=\[(\d+)\]`)
+)
+
+func checkDER(sa core.StorageAuthority, der []byte) error {
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return fmt.Errorf("Failed to parse DER: %s", err)
+	}
+	_, err = sa.GetCertificate(core.SerialToString(cert.SerialNumber))
+	if err != nil && !strings.HasPrefix(err.Error(), "Certificate does not exist for ") {
```

Comment:
Please change GetCertificate to return a special NotFound error so we can avoid string matching on errors.